repoman/pym/repoman/modules/scan/keywords/keywords.py

```diff
@@ -133,6 +135,16 @@ def _checkForMaskLikeKeywords(self, xpkg, y_ebuild, keywords):
 				self.qatracker.add_error("KEYWORDS.stupid",
 					"%s/%s.ebuild" % (xpkg, y_ebuild))
 
+	def _checkForUnsortedKeywords(self, ebuild, xpkg, y_ebuild):
+		'''Ebuilds that contain KEYWORDS
+		which are not sorted alphabetically.'''
+		sorted_keywords = sorted(ebuild.keywords,
+			key=lambda x: x.lstrip("-~"))
```

Comment:
This is broken implementation of sorting.

`ekeyword` tool uses https://gitweb.gentoo.org/proj/gentoolkit.git/tree/pym/gentoolkit/ekeyword/ekeyword.py?id=bb585671418fa51682a7e1dbfef5914e26261adf#n82 which places `*-*` keywords after other keywords, so e.g. `~amd64 ~x86 ~amd64-fbsd ~x86-fbsd ~amd64-linux ~x86-linux` is properly sorted value.

Reply:
Then send in a fix instead of just peanut gallery'ing